agent/bench-scripts/pbench-fio

```diff
@@ -858,16 +865,17 @@ function fio_run_benchmark() {
 								mkdir -p $benchmark_results_dir
 								fio_run_job "$iteration" "$benchmark_results_dir" "$fio_job_file" "$dev" "$clients"
 							else
-								# if we are only postprocessing, then we might have to untar an existing result
-								pushd $iteration_dir >/dev/null
-								if [ ! -e sample$sample ]; then
-									if [ -e sample$sample.tar.xz ]; then
+								# we are only postprocessing, so we might have to untar an existing result
+								if [ ! -e $benchmark_results_dir ]; then
+									if [ -e ${iteration_dir}/sample${sample}.tar.xz ]; then
```

Comment:
I concur with the idea in general; however, in this case, in the next line we change the working directory to `$iteration_dir`, and in the line after that we reference `sample$sample.tar.xz`, so it makes reasonable sense to reference them both at line 870, rather than relying on the combination of them that we previously ensconced in `$benchmark_results_dir`.

Reply:
I don't find `$(dirname ${benchmark_result_dir})` to be easier to read or execute than `$iteration_dir`; likewise I don't think that `$(basename ${benchmark_result_dir}).tar.xz` is better than `sample${sample}.tar.xz`.

Under other circumstances, we might be inclined to cache the values returned by `dirname` and `basename` in variables, so, since we already have them, why not use them?

Reply:
Okay, that is fair given you are not making this any worse than it was, and hopefully we can drop the code entirely in the future.